x-pack/plugin/repository-encrypted/src/internalClusterTest/java/org/elasticsearch/repositories/encrypted/EncryptedRepositorySecretIntegTests.java

```diff
@@ -420,7 +420,16 @@ public void testSnapshotIsPartialForMissingPassword() throws Exception {
             incompleteSnapshotResponse.getSnapshotInfo()
                 .shardFailures()
                 .stream()
-                .allMatch(shardFailure -> shardFailure.reason().contains("[" + repositoryName + "] missing"))
+                .allMatch(
+                    shardFailure -> shardFailure.reason()
+                        .contains(
+                            "RepositoryException[["
+                                + repositoryName
+                                + "] failed to create repository]; nested: IllegalArgumentException[Secure setting [repository.encrypted."
+                                + repositoryName
+                                + ".password] must be set]"
+                        )
```

Comment:
I think this over-constrains the message, really the only important bit is to report the problem about the setting:

```suggestion
                    shardFailure -> shardFailure.reason()
                        .contains("Secure setting [repository.encrypted." + repositoryName + ".password] must be set")
```